agent/xds/clusters.go

```diff
@@ -262,55 +273,36 @@ func (s *Server) clustersFromSnapshotIngressGateway(cfgSnap *proxycfg.ConfigSnap
 	return clusters, nil
 }
 
-func (s *Server) makeAppCluster(cfgSnap *proxycfg.ConfigSnapshot, name, pathProtocol string, port int) (*envoy.Cluster, error) {
-	var c *envoy.Cluster
-	var err error
-
-	cfg, err := ParseProxyConfig(cfgSnap.Proxy.Config)
-	if err != nil {
-		// Don't hard fail on a config typo, just warn. The parse func returns
-		// default config if there is an error so it's safe to continue.
```

Comment:
I think this comment may have been a lie. The error is eventually returned from this function at the bottom of the function. I'm not sure if that was the intent or not.

What is the correct behavior here? Why is it ok to ignore configuration errors?